src/main/java/org/broadinstitute/hellbender/tools/htsgetreader/HtsgetResponse.java

```diff
@@ -53,26 +45,18 @@ public HtsgetClass getDataClass() {
         }
 
         /**
-         * Returns InputStream containing data of a single block from the htsget response
-         *
-         * Large blocks (those behind an http(s) URI) are first saved
-         * to a temp file that is deleted upon program exit
+         * Gets data from this block either from its base64 encoded data url or by making an http request
+         * @return InputStream of data from this block
          */
         public InputStream getData() {
             switch (this.getUri().getScheme()) {
                 case "http":
                 case "https":
                     final HttpGet get = new HttpGet(this.getUri());
                     this.getHeaders().forEach(get::addHeader);
-                    try (final CloseableHttpResponse resp = HttpUtils.getClient().execute(get)) {
-                        final Path outputFile = IOUtils.createTempPath("htsget-temp", "");
-                        try (final OutputStream ostream = Files.newOutputStream(outputFile);
-                             final InputStream istream = resp.getEntity().getContent()) {
-                            org.apache.commons.io.IOUtils.copy(istream, ostream);
-                        } catch (final IOException e) {
-                            throw new UserException("Could not write to temp file", e);
-                        }
-                        return Files.newInputStream(outputFile);
+                    try {
+                        final HttpResponse resp = HttpUtils.getClient().execute(get);
+                        return new AutoCloseInputStream(resp.getEntity().getContent());
```

Comment:
I've always sort of thought this is how our closeable iterators should behave too... Maybe we should change that.